kube-client/src/config/mod.rs

```diff
@@ -154,6 +155,8 @@ pub struct Config {
     ///
     /// If not set, the `cluster_url` is used instead
     pub tls_server_name: Option<String>,
+    /// Headers to pass with every request.
+    pub headers: Vec<(HeaderName, HeaderValue)>,
```

Comment:
Do you think maybe we should call this `extra_headers` to avoid confusion about whether this is something mandatory?

Reply:
Most of the struct fields here are optional, API (new, try etc) don't accept headers so it'd be pretty weird if somebody somehow thinks this is mandatory (if `Config::new` had it as a parameter I would've agreed though..)